vnext/Directory.Build.targets

```diff
@@ -10,6 +10,7 @@
     <Message Text="=> PlatformTarget          [$(PlatformTarget)]" />
     <Message Text="=> PlatformName            [$(PlatformName)]" />
     <Message Text="=> DefaultPlatformToolset  [$(DefaultPlatformToolset)]" />
+    <Message Text="=> ReactNativeWindowsBuildDir  [$(ReactNativeWindowsBuildDir)]" />    
```

Comment:
This variable should not be needed in the MSBuild files.
It should be enough to override `BaseIntDir` and `BaseOutDir` like follows:

```
/p:BaseIntDir=$(ReactNativeWindowsBuildDir)\build\
/p:BaseOutDir=$(ReactNativeWindowsBuildDir)\target\
```

Reply:
I will not go /p:BaseIntDir. I think my solution is more readable:
1. BaseIntDir will be more complex.
I need to change
```
    <BaseOutDir Condition="'$(Platform)' != 'Win32'">$(ReactNativeWindowsDir)target\$(Platform)\$(Configuration)</BaseOutDir>
```
To
```
    <BaseOutDir Condition="'$(Platform)' != 'Win32' and '$(BaseOutDir)' == ''">$(ReactNativeWindowsDir)target\$(Platform)\$(Configuration)</BaseOutDir>
```

2. I still want to keep the platform and configuration in the build output.
It will be easy to compare the buildoutput with local build.

Reply:
> I will not go /p:BaseIntDir

And you don't have to. For local development, these variables don't need to be changed or even known at all.
We should, though, use the available code for automation and not introduce additional redundant constructs.

`Base[Int|Out]Dir` is consistent with the MSBuild predefined variables:
https://docs.microsoft.com/en-us/cpp/build/reference/common-macros-for-build-commands-and-properties?view=vs-2019
Both names are as readable as any MSBuild common property.

> BaseIntDir will be more complex.

Not exactly. It serves a different purpose. We want the flexibility of potentially setting the intermediate and output directories in different locations (think of drive space optimizations, or simpler artifact directory deployment, for instance). The variable you are introducing would serve as a common output root.

`ReactNativeWindowsBuildDir` is still lengthy. And redundant. We need to reduce the amount of build variables prefixed `ReactNativeWindows` or `RNW` where such distinction is not necessary.
It would suffice to call it `BuildRoot`, indicating the build artifact hierarchy will be contained in this directory.

Reply:
I renamed ReactNativeWindowsBuildDir  to BaseIntDir